game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua

```diff
@@ -0,0 +1,686 @@
+									caster:GetAbsOrigin(),
+									nil,
+									50000,
+									DOTA_UNIT_TARGET_TEAM_ENEMY,
+									DOTA_UNIT_TARGET_HERO + DOTA_UNIT_TARGET_BASIC,
+									DOTA_UNIT_TARGET_FLAG_NOT_ANCIENTS,
+									FIND_ANY_ORDER,
+									false)
+	
+	for _,enemy in pairs(enemies) do
+		if enemy.hit_by_earth_spikes then
+			enemy.hit_by_earth_spikes = nil
+		end
+		
+		if enemy.earth_spikes_incoming then
+			enemy.earth_spikes_incoming = nil
+		end
+	end	
+	
+	
+	-- Decide if spell was cast on ground or on a target unit
+	if not target then
+		target = target_point
+	else
+		target = target:GetAbsOrigin()
+	end
+	
+	-- Adjust length if caster has cast increase
+	travel_distance = travel_distance + GetCastRangeIncrease(caster)
+	
+		
+	-- Play sound
+	caster:EmitSound(sound_cast)		
+	
+	-- Launch line projectile
+	local spikes_projectile = {	Ability = ability,
+								EffectName = particle_projectile,
+								vSpawnOrigin = caster:GetAbsOrigin(),
+								fDistance = travel_distance,
+								fStartRadius = spikes_radius,
+								fEndRadius = spikes_radius,
+								Source = caster,
+								bHasFrontalCone = false,
+								bReplaceExisting = false,
+								iUnitTargetTeam = DOTA_UNIT_TARGET_TEAM_ENEMY,							
+								iUnitTargetType = DOTA_UNIT_TARGET_HERO + DOTA_UNIT_TARGET_BASIC,							
+								bDeleteOnHit = false,
+								vVelocity = (target - caster:GetAbsOrigin()):Normalized() * spike_speed,
+								bProvidesVision = false,							
+							}
+							
+	ProjectileManager:CreateLinearProjectile(spikes_projectile)
+	
+end
+
+function EarthSpikesHit ( keys )
+	-- Ability properties
+	local caster = keys.caster
+	local target = keys.target	
+	local ability = keys.ability
+	local ability_level = ability:GetLevel()-1
+	local sound_impact = keys.sound_impact
+	local particle_hit = keys.particle_hit	
+	local sound_cast = keys.sound_cast
+	local particle_projectile = keys.particle_projectile
+	
+	-- Ability specials	
+	local knock_up_height = ability:GetLevelSpecialValueFor("knock_up_height", ability_level)
+	local knock_up_time = ability:GetLevelSpecialValueFor("knock_up_time", ability_level)
+	local damage = ability:GetLevelSpecialValueFor("damage", ability_level)
+	local stun_duration = ability:GetLevelSpecialValueFor("stun_duration", ability_level)
+	local spike_speed = ability:GetLevelSpecialValueFor("spike_speed", ability_level)
+	local travel_distance = ability:GetLevelSpecialValueFor("travel_distance", ability_level)
+	local spikes_radius = ability:GetLevelSpecialValueFor("spikes_radius", ability_level)	
+	local extra_spike_AOE = ability:GetLevelSpecialValueFor("extra_spike_AOE", ability_level)
+	local wait_interval = ability:GetLevelSpecialValueFor("wait_interval", ability_level)
+			
+	-- Spikes ignore target if it was already hit this cast
+	if target.hit_by_earth_spikes then
+		return nil
+	end
+	
+	-- Mark target as hit by Earth Spikes
+	target.hit_by_earth_spikes = true
```

Comment:
This is a concept that he asked for. I've built IMBA Lion with his instructions of how the skills should work and interact.

The Skywrath Mage concept was made by me entirely though, as a test for myself whether I could develop a hero from scratch.

Shush